include/class.file.php

```diff
@@ -358,6 +356,10 @@ static function uploadLogo($file, &$error, $aspect_ratio=2) {
     }
 
     static function create(&$file, $ft='T', $deduplicate=true) {
+        // Crap-out if file contents do not match expected file type
+        if (strpos('image/', $ft) !== 0 && !exif_imagetype($file['tmp_name']))
```

Comment:
`$ft` is wrong where - it should be `$file['type']`. And on the second thought - this check should go to FileUploadField.